dev/tests/api-functional/testsuite/Magento/GraphQl/SendFriend/SendFriendTest.php

```diff
@@ -0,0 +1,469 @@
+    }
+
+    public function testSendWithoutExistProduct()
+    {
+        $query =
+            <<<QUERY
+mutation {
+    sendEmailToFriend(
+        input: {
+          product_id: 2018
+          sender: {
+            name: "Name"
+            email: "e@mail.com"
+            message: "Lorem Ipsum"
+        }          
+          recipients: [
+              {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+              {
+                  name: "Recipient Name 2"
+                  email:"recipient2@mail.com"
+              }
+          ]
+        } 
+    ) {
+        sender {
+            name
+            email
+            message
+        }
+        recipients {
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage('The product that was requested doesn\'t exist. Verify the product and try again.');
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     */
+    public function testMaxSendEmailToFriend()
+    {
+        /** @var SendFriend $sendFriend */
+        $sendFriend = $this->sendFriendFactory->create();
+
+        $query =
+            <<<QUERY
+mutation {
+    sendEmailToFriend(
+        input: {
+          product_id: 1
+          sender: {
+            name: "Name"
+            email: "e@mail.com"
+            message: "Lorem Ipsum"
+        }
+          recipients: [
+                {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+               {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+               {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+               {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+               {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               },
+               {
+                  name: "Recipient Name 1"
+                  email:"recipient1@mail.com"
+               }
+          ]
+        }
+    ) {
+        sender {
+            name
+            email
+            message
+        }
+        recipients {
+            name
+            email
+        }
+    }
+}
+QUERY;
+        $this->expectException(\Exception::class);
+        $this->expectExceptionMessage("No more than {$sendFriend->getMaxRecipients()} emails can be sent at a time.");
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/SendFriend/_files/product_simple.php
+     */
+    public function testSendWithoutRecipentsName()
```

Comment:
Looks like for validation w can use `dataProviders` instead of creating separate methods